pkg/server/server.go

```diff
@@ -430,7 +433,7 @@ func (s *ProxyServer) serveRecvFrontend(stream client.ProxyService_ProxyServer,
 			// the address, then we can send the Dial_REQ to the
 			// same agent. That way we save the agent from creating
 			// a new connection to the address.
-			backend, err = s.getBackend(pkt.GetDialRequest().Address)
+			backend, agentID, err = s.getBackend(pkt.GetDialRequest().Address)
```

Comment:
Here is an example of why I do not think we should return a destHost as a backed identifier.